### ISSUE TITLE
Super fast?

### ISSUE THREAD

**Issue by cflaviu** [felixguendling/cpp-serialization-benchmark#4]
I have run the benchmarks and cista_offset_slim_bench serialization looks [super fast ](https://gist.github.com/cflaviu/657848f15c9665f723d683d5134585d9). Could this serialization be 14 times faster than cap'n proto comparing to your result where it was 1.4 times faster than cap'n proto? Or is there a glitch? I didn't debugged it.

**Comment by felixguendling** [comment]
I would assume that this heavily depends on the dataset size vs. CPU cache size. As you ran the benchmark on a CPU with 65536K L3 cache, the dataset fits completely into the cache. I benchmarked with a end-2013 Mac Book Pro with a much smaller cache size. I would assume that Cista benefits more from changes in cache size than Cap'n'Proto. But that's just a guess. Additionally, maybe the compiler was able to optimize more. I used Apple Clang, you used probably GCC.

**Comment by cflaviu** [comment]
Indeed, L3 cache could hold all the data and this is a good reason for such performance. I used Clang 11 and GCC 10 getting similar results with -O3 flag. I tried to use -march=znver2 optimization flag but build failed due invalid instruction generation (segmentation fault). Maybe this is because I'm compiling inside a Windows hosted Linux VM (VirtualBox). Therefore the generated binary was not fully optimized.
